api/src/main/java/org/jenkinsci/plugins/workflow/graph/FlowGraphWalker.java

```diff
@@ -13,8 +15,7 @@
  * @author Kohsuke Kawaguchi
  */
 // TODO guarantee DFS order (is it not already in DFS order?)
```

Comment:
I think all this new code is already covered by existent tests. Let's leave this test for later.